cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/Node.kt

```diff
@@ -257,6 +257,12 @@ abstract class Node :
     @Relationship("ANNOTATIONS") var annotationEdges = astEdgesOf<Annotation>()
     var annotations by unwrapping(Node::annotationEdges)
 
+    /**
+     * Additional problem nodes. These nodes represent problems which occurred during processing of
+     * a node (i.e. only partially processed).
+     */
+    val additionalProblems: MutableSet<ProblemNode> = mutableSetOf()
```

Comment:
I wonder if it would make sense to make this an AST edge. But in this case we need to either use `ProblemDeclaration` or `ProblemExpression` since our AST edges only work with a real `Node` and `ProblemNode` is an interface (although misleadingly named).

Reply:
The advantage of making this an AST edge would be that we could gather all problems using `allChildren` at some point.

Reply:
> But it's not really part of the AST, so I wouldn't use this hack. At some point, we'll just remove it and we'll be surprised why things are missing ;) But you have a point: There there's a shortcut to collect problems (https://github.com/Fraunhofer-AISEC/cpg/blob/main/cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/Extensions.kt#L608), and I guess we have to add this set to the result there. I'm wondering if we have to do this recursively though...

We probably can't, since it uses `allChildren` and that only uses AST edges.

Reply:
> > But it's not really part of the AST, so I wouldn't use this hack. At some point, we'll just remove it and we'll be surprised why things are missing ;) But you have a point: There there's a shortcut to collect problems (https://github.com/Fraunhofer-AISEC/cpg/blob/main/cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/Extensions.kt#L608), and I guess we have to add this set to the result there. I'm wondering if we have to do this recursively though...
> 
> We probably can't, since it uses `allChildren` and that only uses AST edges.

Still, I'm with @KuechA on this. It's a stretch to say theses nodes are part of the AST. If we have them as AST nodes now, then we should at least try to come up with a cleaner solution asap.

Reply:
After internal discussion, we keep those problems out of the AST, but regards this as a *temporary* solution and have a look at a cleaner design that encompasses all problems (AST translation, parsing, passes).